platform/darwin/src/NSPredicate+MGLAdditions.mm

```diff
@@ -294,13 +294,39 @@ + (instancetype)mgl_predicateWithJSONObject:(id)object {
         return [NSPredicate predicateWithValue:YES];
     }
     if ([op isEqualToString:@"all"]) {
-        NSArray *subpredicates = MGLSubpredicatesWithJSONObjects([objects subarrayWithRange:NSMakeRange(1, objects.count - 1)]);
+        NSArray *jsonObjects = [objects subarrayWithRange:NSMakeRange(1, objects.count - 1)];
+        NSArray *subpredicates = MGLSubpredicatesWithJSONObjects(jsonObjects);
+        if (jsonObjects.count == 2) {
+            // Determine if the expression is of BETWEEN type
+            if ([jsonObjects[0] isKindOfClass:[NSArray class]] &&
+                [jsonObjects[1] isKindOfClass:[NSArray class]]) {
+                NSArray *leftCondition = jsonObjects[0];
+                NSArray *rightCondition = jsonObjects[1];
+                NSString *leftOperator = leftCondition.firstObject;
+                NSString *rightOperator = rightCondition.firstObject;
+                if ([leftOperator isEqualToString:@"<="] && [rightOperator isEqualToString:@"<="]) {
+                    return [NSPredicate predicateWithFormat:@"%@ BETWEEN %@", [NSExpression mgl_expressionWithJSONObject:leftCondition[2]], @[[NSExpression mgl_expressionWithJSONObject:leftCondition[1]], [NSExpression mgl_expressionWithJSONObject:rightCondition[2]]]];
+                } else if([leftOperator isEqualToString:@">="] && [rightOperator isEqualToString:@"<="]) {
+                    return [NSPredicate predicateWithFormat:@"%@ BETWEEN %@", [NSExpression mgl_expressionWithJSONObject:leftCondition[1]], @[[NSExpression mgl_expressionWithJSONObject:leftCondition[2]], [NSExpression mgl_expressionWithJSONObject:rightCondition[2]]]];
+                } else if([leftOperator isEqualToString:@">="] && [rightOperator isEqualToString:@">="]) {
+                    return [NSPredicate predicateWithFormat:@"%@ BETWEEN %@", [NSExpression mgl_expressionWithJSONObject:leftCondition[1]], @[[NSExpression mgl_expressionWithJSONObject:leftCondition[2]], [NSExpression mgl_expressionWithJSONObject:rightCondition[1]]]];
+                }
+            }
+        }
         return [NSCompoundPredicate andPredicateWithSubpredicates:subpredicates];
     }
     if ([op isEqualToString:@"any"]) {
         NSArray *subpredicates = MGLSubpredicatesWithJSONObjects([objects subarrayWithRange:NSMakeRange(1, objects.count - 1)]);
         return [NSCompoundPredicate orPredicateWithSubpredicates:subpredicates];
     }
+    if ([op isEqualToString:@"match"]) {
+        NSArray *subpredicates = MGLSubexpressionsWithJSONObjects([objects subarrayWithRange:NSMakeRange(1, objects.count - 1)]);
+        return [NSPredicate predicateWithFormat:@"%@ IN %@" argumentArray:subpredicates];
+    }
+    if ([op isEqualToString:@"has"]) {
```

Comment:
A [`has`](https://www.mapbox.com/mapbox-gl-js/style-spec/#expressions-has) expression checks whether an object has a particular property; it doesn’t check whether an array contains a particular object. We need to convert a `match` expression to either the `CONTAINS` or `IN` operator (pick one), but we shouldn’t convert a `has` expression to either operator.